src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -230,13 +236,33 @@ public Profile getProfile(String email) {
 
     // (String) profileEntity.getProperty("profilePicURL") redacted
     Profile profile = new Profile((String) profileEntity.getProperty("email"),
-        (String) profileEntity.getProperty("name"), 
-        (Double) profileEntity.getProperty("latitude"),
+        (String) profileEntity.getProperty("name"), (Double) profileEntity.getProperty("latitude"),
         (Double) profileEntity.getProperty("longitude"),
-        (String) profileEntity.getProperty("phone"),(String) 
-        profileEntity.getProperty("schedule"));
+        (String) profileEntity.getProperty("phone"),
+        (String) profileEntity.getProperty("schedule"));
+
+    return profile;
+  }
+
+  /**
+   * Returns the Item owned by the email address, or null if no matching Item was found. TODO: an
+   * item is uniquely identified by an email. change later so it is by a unique random ID
+   */
 
-    return profile;   
+  public Item getPosting(String email) {
+    
+    Query query = new Query("Posting")
+        .setFilter(new Query.FilterPredicate("email", FilterOperator.EQUAL, email));
+    PreparedQuery results = datastore.prepare(query);
+    Entity itemEntity = results.asSingleEntity();
+    if (itemEntity == null) {
+      return null;
```

Comment:
I'm guessing if itemEntity is null it's an error? Might be worth throwing here, or at least logging something on the server side so we know something happened.

The hardest to find bugs are when something like this occurs and the server silently ignores it.

Reply:
Fair! fixing this for sure.